### PR TITLE
Windows build script errorlevel fixes

### PR DESCRIPTION
**Pull request by amzn-changml**
## What does this PR do?

A number of fixes to resolve issues found in: https://github.com/o3de/o3de/issues/17791

- Changes the errorlevel variable in the Windows build script to enforce expansion
- Changes Windows build errorlevel comparators to numeric

## How was this PR tested?

Tested in sandbox Jenkins

### CODE REVIEW

File: scripts/build/Platform/Windows/build_windows.cmd [hunk 1]
```diff
@@ -50,15 +50,15 @@ IF NOT EXIST CMakeCache.txt (
...
     ECHO !CONFIGURE_CMD!> %LAST_CONFIGURE_CMD_FILE%
 )
 
 REM Split the configuration on semi-colon and use the cmake --build wrapper to run the underlying build command for each
 FOR %%C in (%CONFIGURATION%) do (
-    call ECHO [ci_build] cmake --build . --target %CMAKE_TARGET% --config %%C %CMAKE_BUILD_ARGS% -- %CMAKE_NATIVE_BUILD_ARGS%
-    call cmake --build . --target %CMAKE_TARGET% --config %%C %CMAKE_BUILD_ARGS% -- %CMAKE_NATIVE_BUILD_ARGS%
-    IF NOT %ERRORLEVEL%==0 GOTO :error
+    call ECHO [ci_build] cmake --build . --target %CMAKE_TARGET% --config %%C %CMAKE_BUILD_ARGS% -- %CMAKE_NATIVE_BUILD_ARGS% -j %HALF_PROCESSORS%
+    call cmake --build . --target %CMAKE_TARGET% --config %%C %CMAKE_BUILD_ARGS% -- %CMAKE_NATIVE_BUILD_ARGS% -j %HALF_PROCESSORS%
```
[the remark this answers is not in the document]
Comment (amzn-changml) [reply scripts/build/Platform/Windows/build_windows.cmd]: I actually changed it for experimentation and now reverting it. The issue is that the Android profile build uses `build_windows.bat` which also pulls in the `-j` parameter for ninja. Setting ` /p:CL_MPCount` is incompatible with ninja as well as shown in this run: https://jenkins.build.o3de.org/blue/rest/organizations/jenkins/pipelines/O3DE/branches/PR-17794/runs/1/nodes/583/log/?start=0

[the remark this answers is not in the document]
Comment (spham-amzn) [reply scripts/build/Platform/Windows/build_windows.cmd]: Oh, that's not good. I'll create a GHI to create a new Ninja build pipeline, as part of another Android legacy deprecation that I'll create as well

**Review by nick-l-o3de — APPROVED** [review]

File: scripts/build/Platform/Windows/build_windows.cmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (nick-l-o3de) [reply scripts/build/Platform/Windows/build_windows.cmd]: You don't have to specify any -j for ninja, afaik.  its defaulted to a reasonable number last I checked

**Comment by amzn-changml** [comment]
I should mention that this PR did detect the C2220 error as warning as called out by: https://github.com/o3de/o3de-extras/issues/687. I've since fixed the error: https://github.com/o3de/o3de-extras/pull/688, so this next run should pass

File: scripts/build/Platform/Windows/build_windows.cmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (amzn-changml) [reply scripts/build/Platform/Windows/build_windows.cmd]: Going to separate parallelization limits to a new PR for now. We'll keep the scope of this PR to the errorlevel handling

**Review by lemonade-dm — APPROVED** [review]

**Pull request merged**